client/llb/source.go

```diff
@@ -116,6 +116,11 @@ func Image(ref string, opts ...ImageOption) State {
 		attrs[pb.AttrImageRecordType] = info.RecordType
 	}
 
+	if ll := info.layerLimit; ll != nil {
+		attrs[pb.AttrImageLayerLimit] = strconv.FormatInt(int64(*ll), 10)
```

Comment:
Probably worth setting this as a uint or just enforcing that it's not negative.

Reply:
In `attrs` it is a string. I don't want to set `WithLayerLimit` to `uint` as then it will require casting on every usage. I've added validation on the server-side to make sure the value is positive.